### PR TITLE
proto: remove `(uint32_t)-1`

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

### CODE REVIEW

File: proto/rootlesscontainers.proto [hunk 1]
```diff
@@ -25,7 +25,7 @@ package rootlesscontainers;
 message Resource {
    // Zero-value MUST be parsed as a literally zero-value, not "unset".
    // To keep both uid and gid unchaged, the entire xattr value SHOULD be removed.
-   // To keep either one of uid or gid unchaged, -1 value SHOULD be set.
```
Comment (cyphar) [inline proto/rootlesscontainers.proto]: You could make this "0xFFFFFF (in other words, (uint32_t) -1)" to make it clear why that value in particular.

Comment (AkihiroSuda) [reply proto/rootlesscontainers.proto]: done

**Comment by cyphar** [comment]
Oh wait. @AkihiroSuda I just realised that having an "unset" version of `uid` and `gid` doesn't make a lot of sense -- since a value of `0` would be identical to it being unset. Maybe we should just go with Go's zero-value rules here?

**Comment by AkihiroSuda** [comment]
>  since a value of 0 would be identical to it being unset

I believe 0 stands for UID/GID 0.

**Comment by cyphar** [comment]
Right, but in the context of a rootless container, a UID/GID of 0 is the same as a file without a `user.rootlesscontainers` xattr (since root in the container is the normal user on the host).

The reason I'm making a point of this is that in protobuf, integers are variable in size (even `uint32`) and so `0` is represented in fewer bytes than `0xFFFFFFFF` -- not to mention that `0xFFFFFFFF` is slightly more arbitrary than `0` (because `0` is more consistent from my PoV).

**Comment by AkihiroSuda** [comment]
I think it should be valid to use both ptracer and uid/gid mappings, even if it is  practically useless.

If `chown 2001` is invoked against a file owned by `1001:1001`, ptracer is expected to create `{UID: 2001, GID:-1}` xattr (of course `{UID: 2001, GID: 1001`} is also fine).

`{UID: 2001, GID: 0}` xattr is expected for `chown 2001:0`.

**Comment by cyphar** [comment]
Hmm. I guess that's fair enough (though I agree it doesn't make much sense to use the tracer outside of a container).

**Comment by cyphar** [comment]
LGTM.

**Pull request merged**